### PR TITLE
Updates documentation and adds assertion for files so that glob pattern can't include directories.

### LINKED ISSUE #14: Commit a05e4a1 breaks existing clients
I haven't had a chance to look deeper into this, but:

```
Hdfs.files(conf.outputBasePath.toPath, "*/*/*")
```
Now returns no files where it used to succeed

### PR DESCRIPTION
**Pull request by stephanh**
This closes #14.

### CODE REVIEW

File: src/main/scala/au/com/cba/omnia/permafrost/hdfs/Hdfs.scala [hunk 1]
```diff
@@ -81,8 +86,8 @@ object Hdfs extends ResultantOps[Hdfs] with ToResultantMonadOps {
     } yield files.sortBy(_.toString)
```
Comment (rowandavies) [inline src/main/scala/au/com/cba/omnia/permafrost/hdfs/Hdfs.scala]: format: extra space before `<-` in the lines above (existing issue)

File: src/main/scala/au/com/cba/omnia/permafrost/hdfs/Hdfs.scala [hunk 2]
```diff
@@ -58,10 +58,15 @@ object Hdfs extends ResultantOps[Hdfs] with ToResultantMonadOps {
...
-   */
-  def files(dir: Path, globPattern: String = "*", recursive: Boolean = false) = {
+  /**
+    * Lists all files/directories matching `globPattern` under `dir` path, sorted by name.
+    * 
+    * Includes results from subdirectories, breadth first, if 'recursive' is true.
+    * NB: `globPattern` only applies to children files/directories. `* / *.txt` is not a valid
+    * glob since it goes depper than the child files/directores.
+    */
+  def files(dir: Path, globPattern: String = "*", recursive: Boolean = false): Hdfs[List[Path]] = {
```
Comment (rowandavies) [inline src/main/scala/au/com/cba/omnia/permafrost/hdfs/Hdfs.scala]: extension: For the hdfs-purge script, we used a version of files that is depth first and produces a delayed stream that can be consumed tail-recursively in bounded space via forEach, foldLeft, etc.   Probably this should stay as a separate method, but perhaps it belongs here in permafrost, and we could implement `files` mostly just by converting the stream to a list.

File: src/main/scala/au/com/cba/omnia/permafrost/hdfs/Hdfs.scala (hunk 1 above)
Comment (stephanh) [inline src/main/scala/au/com/cba/omnia/permafrost/hdfs/Hdfs.scala]: Fixed.

File: src/main/scala/au/com/cba/omnia/permafrost/hdfs/Hdfs.scala [hunk 3]
```diff
@@ -58,12 +58,17 @@ object Hdfs extends ResultantOps[Hdfs] with ToResultantMonadOps {
...
-  /** List all files/directories matching `globPattern` under `dir` path, sorted by name.
-   *  Include results from subdirectories, breadth first, if 'recursive' is true.
-   */
-  def files(dir: Path, globPattern: String = "*", recursive: Boolean = false) = {
+  /**
+    * Lists all files/directories matching `simpleGlob` under `dir` path, sorted by name.
+    * 
+    * Includes results from subdirectories, breadth first, if 'recursive' is true.
+    * NB: `simpleGlob` only applies to children files/directories. `* / *.txt` is not a valid
+    * glob since it goes depper than the child files/directores.
```
Comment (vineethvarghese) [inline src/main/scala/au/com/cba/omnia/permafrost/hdfs/Hdfs.scala]: deeper

File: src/main/scala/au/com/cba/omnia/permafrost/hdfs/Hdfs.scala [hunk 4]
```diff
@@ -58,12 +58,17 @@ object Hdfs extends ResultantOps[Hdfs] with ToResultantMonadOps {
...
     Hdfs.filesystem.map(fs => f(fs))
 
-  /** List all files/directories matching `globPattern` under `dir` path, sorted by name.
-   *  Include results from subdirectories, breadth first, if 'recursive' is true.
-   */
-  def files(dir: Path, globPattern: String = "*", recursive: Boolean = false) = {
+  /**
+    * Lists all files/directories matching `simpleGlob` under `dir` path, sorted by name.
+    * 
+    * Includes results from subdirectories, breadth first, if 'recursive' is true.
```
Comment (vineethvarghese) [inline src/main/scala/au/com/cba/omnia/permafrost/hdfs/Hdfs.scala]: Does mentioning that we do a breath first search have any impact on the user given that we sort by name. We may want to mention the fact that recursive could lead to a huge in memory list.

**Comment by rowandavies** [comment]
Looks good to me.  :+1:

**Comment by vineethvarghese** [comment]
Have a few comments but otherwise :+1:

File: src/main/scala/au/com/cba/omnia/permafrost/hdfs/Hdfs.scala (hunk 4 above)
Comment (stephanh) [inline src/main/scala/au/com/cba/omnia/permafrost/hdfs/Hdfs.scala]: I changed the comment and added the warning about stream based. I'll let whoever has the stream based version raise a PR for their result.

File: src/main/scala/au/com/cba/omnia/permafrost/hdfs/Hdfs.scala (hunk 3 above)
Comment (stephanh) [inline src/main/scala/au/com/cba/omnia/permafrost/hdfs/Hdfs.scala]: Fixed. Thank you.

**Pull request merged**